lib/hardware/modules/iob_mux/iob_mux.py

```diff
@@ -51,6 +56,23 @@ def setup(py_params_dict):
                 ],
             },
         ],
+        "snippets": [
+            {
+                "verilog_code": """
+            reg [DATA_W-1:0] data_o_reg;
+            assign data_o = data_o_reg;
+            integer input_sel;
+            always @* begin
+                data_o_reg = {DATA_W{1'b0}};
+                for (input_sel = 0; input_sel < N; input_sel = input_sel + 1) begin : gen_mux
+                    if (input_sel == sel_i) begin
+                         data_o_reg = data_i[input_sel*DATA_W+:DATA_W];
+                    end
+                end
+            end    
```

Comment:
The new py2hwsw version 0.7.5, automatically detects output ports of type "reg". So there is no need to create a new internal `data_o_reg` for this module.
```suggestion
            integer input_sel;
            always @* begin
                data_o = {DATA_W{1'b0}};
                for (input_sel = 0; input_sel < N; input_sel = input_sel + 1) begin : gen_mux
                    if (input_sel == sel_i) begin
                         data_o = data_i[input_sel*DATA_W+:DATA_W];
                    end
                end
            end    
```